src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationSource.cs

```diff
@@ -11,21 +13,44 @@ internal class AzureAppConfigurationSource : IConfigurationSource
         private readonly bool _optional;
         private readonly Func<AzureAppConfigurationOptions> _optionsProvider;
         private readonly IConfigurationClientFactory _configurationClientFactory;
+        private readonly string _environmentName;
 
-        public AzureAppConfigurationSource(Action<AzureAppConfigurationOptions> optionsInitializer, bool optional = false, IConfigurationClientFactory configurationClientFactory = null)
+        public AzureAppConfigurationSource(Action<AzureAppConfigurationOptions> optionsInitializer, string environmentName = null, bool optional = false, IConfigurationClientFactory configurationClientFactory = null)
         {
             _optionsProvider = () => {
                 var options = new AzureAppConfigurationOptions();
                 optionsInitializer(options);
                 return options;
             };
 
+            _environmentName = environmentName;
             _optional = optional;
             _configurationClientFactory = configurationClientFactory ?? new ConfigurationClientFactory();
         }
 
         public IConfigurationProvider Build(IConfigurationBuilder builder)
         {
+            if (_environmentName != null)
+            {
+                try
+                {
+                    string envType = Environment.GetEnvironmentVariable(RequestTracingConstants.AspNetCoreEnvironmentVariable) ??
+                                        Environment.GetEnvironmentVariable(RequestTracingConstants.DotNetCoreEnvironmentVariable);
+                    if (!envType.Equals(_environmentName, StringComparison.OrdinalIgnoreCase))
+                    {
+                        return new EmptyConfigurationProvider();
+                    }
+                }
+                catch (SecurityException)
+                {
+                    // Can't read environment variables - assume production environment
+                    if (!_environmentName.Equals(Environments.Production, StringComparison.OrdinalIgnoreCase))
+                    {
+                        return new EmptyConfigurationProvider();
```

Comment:
This will be a silent failure. In a way, I think we should throw unless optional is true. But I really hope we can avoid reading environment variables altogether.